test/test-settings.c

```diff
@@ -188,64 +237,145 @@ void taskEntryB(void *aParams)
 
         idx++;
     }
+
+    NL_LOG_CRIT(lrTEST, "'%s' exiting\n", name);
 }
 
-int main(int argc, char **argv)
+static bool is_testing(volatile const taskData_t *aSubscriber,
+                       volatile const taskData_t *aPublisher)
 {
-    int             idx;
-    int             status;
-    int             retval;
-    nl_event_t      *queuemem[50];
-    nleventqueue_t  queue;
+    bool retval = false;
 
-    NL_LOG_CRIT(lrTEST, "start main\n");
+    if ((!aSubscriber->mFailed && !aPublisher->mFailed) &&
+        (!aSubscriber->mSucceeded && !aPublisher->mSucceeded))
```

Comment:
If any one of those four is true, we return false.  Is that what's desired here, or do we want something like this:

    if ((!aSubscriber->mFailed && !aSubscriber->mSucceeded) ||
        (!aPublisher->mFailed  && !aPublisher->mSucceeded))

Reply:
I think there is a bug there; however, I believe this is what is actually desired:

/**
 *  Determine whether or not the main thread should continue to wait
 *  for the test to complete.
 *
 *  The main thread should wait for testing until:
 *
 *    * Either thread failed (mFailed == true).
 *    * Both threads succeeded (mSucceeded == true).
 *
 */
static bool is_testing(volatile const taskData_t *aSubscriber,
                       volatile const taskData_t *aPublisher)
{
    bool retval = true;

    if ((aSubscriber->mFailed || aPublisher->mFailed) ||
        (aSubscriber->mSucceeded && aPublisher->mSucceeded))
    {
        retval = false;
    }

    return retval;
}